pkg/controllers/leasegarbagecollection/suite_test.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/aws/karpenter-core/pkg/apis/settings"
+	"github.com/aws/karpenter-core/pkg/controllers/leasegarbagecollection"
+	"github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter-core/pkg/operator/scheme"
+	"github.com/aws/karpenter-core/pkg/test"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "knative.dev/pkg/logging/testing"
+
+	. "github.com/aws/karpenter-core/pkg/test/expectations"
+)
+
+var ctx context.Context
+var env *test.Environment
+var garbageCollectionController controller.Controller
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "LeaseGarbageCollection")
+}
+
+var _ = BeforeSuite(func() {
+	env = test.NewEnvironment(scheme.Scheme, test.WithCRDs(apis.CRDs...))
+	ctx = settings.ToContext(ctx, test.Settings())
+
+	garbageCollectionController = leasegarbagecollection.NewController(env.Client)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = AfterEach(func() {
+	ExpectCleanedUp(ctx, env.Client)
+})
+
+var _ = Describe("GarbageCollection", func() {
+	Context("Metrics", func() {
+		It("should fire the terminationSummary metric when deleting nodes", func() {
+			badLease := &coordinationsv1.Lease{
+				ObjectMeta: v1.ObjectMeta{
+					CreationTimestamp: v1.Time{Time: time.Now().Add(-time.Hour * 2)},
+					Name:              "new-lease",
+					Namespace:         "kube-node-lease",
+					Labels:            map[string]string{test.DiscoveryLabel: "unspecified"},
+				},
+			}
+			ExpectApplied(ctx, env.Client, badLease)
+			ExpectReconcileSucceeded(ctx, garbageCollectionController, client.ObjectKeyFromObject(badLease))
+			ExpectNotFound(ctx, env.Client, badLease)
+
+			m, ok := FindMetricWithLabelValues("karpenter_nodes_leases_deleted", map[string]string{})
+			Expect(ok).To(BeTrue())
+			Expect(lo.FromPtr(m.GetCounter().Value)).To(BeNumerically("==", 1.0))
+
+		})
+	})
+	It("should not delete node lease that contains an OwnerReference", func() {
+		node := test.Node(test.NodeOptions{})
+		ExpectApplied(ctx, env.Client, node)
+		node = ExpectExists(ctx, env.Client, node)
+		goodLease := &coordinationsv1.Lease{
+			ObjectMeta: v1.ObjectMeta{
+				CreationTimestamp: v1.Time{Time: time.Now().Add(-time.Hour * 2)},
+				Name:              "new-lease",
+				Namespace:         "kube-node-lease",
+				Labels:            map[string]string{test.DiscoveryLabel: "unspecified"},
+				OwnerReferences: []v1.OwnerReference{
+					{
+						APIVersion: "v1",
+						Name:       "owner-node",
+						Kind:       "Node",
+						UID:        node.UID,
+					},
+				},
+			},
+		}
+		ExpectApplied(ctx, env.Client, goodLease)
+		ExpectReconcileSucceeded(ctx, garbageCollectionController, client.ObjectKeyFromObject(goodLease))
+		ExpectExists(ctx, env.Client, goodLease)
+	})
+	It("should delete node lease that does not contain an OwnerReference", func() {
+		badLease := &coordinationsv1.Lease{
```

Comment:
Are any of these declarations repetitive and can be put into a `BeforeEach` statement?